django/http/response.py

```diff
@@ -397,14 +400,39 @@ def __init__(self, *args, as_attachment=False, filename='', **kwargs):
         self.filename = filename
         super().__init__(*args, **kwargs)
 
+    def _wrap_file_to_stream_close(self, filelike):
+        """
+        Wrap the file-like close() with a version that calls
+        FileResponse.close().
+        """
+        closing = False
+        filelike_close = getattr(filelike, 'close', lambda: None)
```

Comment:
Shouldn't close accept `self` since it's a method?

Reply:
No, because this is a "bound" method, which means that `self` is implicitly included. Here is an illustration:

```python
>>> class Filelike:
...     def close(self):
...         return
... 
>>> filelike = Filelike()
>>> close = getattr(filelike, 'close')
>>> close
<bound method Filelike.close of <__main__.Filelike object at 0x1043c5588>>
>>> close(filelike)  # Try passing one argument (doesn't work).
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: close() takes 1 positional argument but 2 were given
>>> close()  # Try passing no arguments (works).
```

Reply:
Sorry, nevermind.